### DISCUSSION TITLE
Pre-filled variable wizzard on manual pipeline run

### DISCUSSION THREAD

**Discussion by wrenix** [woodpecker-ci/woodpecker#4404] [Request a Feature] [answered]
Current manuelle Run Pipeline Wizzard:
![image](https://github.com/user-attachments/assets/ae39e082-a602-431d-b428-d51edbff24ff)

## Suggestion, like gitlab-ci:
![on gitlab](https://about.gitlab.com/images/blogimages/Run-pipeline.gif)
Source https://about.gitlab.com/blog/2020/12/02/pre-filled-variables-feature/

## Doku
  - prefilled Doku: https://docs.gitlab.com/ee/ci/pipelines/index.html#prefill-variables-in-manual-pipelines
  - configuration with `value` or `options` (enum):
https://docs.gitlab.com/ee/ci/yaml/#variablesoptions

**Comment by 6543** [comment]
the biggest question is how should woodpecker determine the variables ...

**Reply by anbraten** [reply]
I would suggest to get it from the pipeline config.

**Reply by 6543** [reply]
As it is rather dynamic ... why not use #3723 as base and add a flag `use for prefill manual pipelines` or so ... ?

**Reply by wrenix** [reply]
In yaml file, it is easier to migrate to a new instance / setup of woodpecker (e.g. if some Organisation migrate between forges)

**Comment by qwerty287** [comment, marked as answer]
I think this is a duplicate of https://github.com/woodpecker-ci/woodpecker/issues/3326?

**Reply by wrenix** [reply]
More or less it is, yes.
(Sorry i have not found it with words like prefilled or wizard).

I Miss there a enum/Option list AS dropdown
